ledger/complete/wal/compactor.go

```diff
@@ -111,26 +114,35 @@ func (c *Compactor) createCheckpoints() (int, error) {
 		return -1, fmt.Errorf("cannot get latest checkpoint: %w", err)
 	}
 
+	// if admin tool was used to trigger checkpoint, then it's equivilent to setting checkpoint distance
+	// as 1 for once.
+	if to-from > 1 && c.triggerCheckpointOnNextSegmentFinish.CAS(true, false) {
+		c.logger.Info().Msgf("manually trigger checkpointing (from: %v, to: %v)", from, to)
```

Comment:
I think `to` segment isn't checkpointed because this code assumes last segment is being written to and not finalized.

```suggestion
		c.logger.Info().Msgf("manually trigger checkpointing (from: %v, to: %v)", from, to-1)
```